runtime/calamari/src/diff_gas_fees.rs

```diff
@@ -0,0 +1,2166 @@
+// Copyright 2020-2023 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(test)]
+
+use crate::{Runtime, TransactionPayment};
+use codec::{Decode, Encode};
+use frame_support::{
+    dispatch::DispatchInfo,
+    traits::{schedule::MaybeHashed, OriginTrait, WrapperKeepOpaque},
+    weights::GetDispatchInfo,
+};
+use manta_primitives::assets::{AssetRegistryMetadata, TestingDefault};
+use manta_support::manta_pay::TransferPost;
+use pallet_transaction_payment::Multiplier;
+use runtime_common::MinimumMultiplier;
+use sp_runtime::{
+    traits::{Hash, Saturating},
+    AccountId32, Perbill, Percent,
+};
+use xcm::prelude::*;
+
+const GAS_FEE_FLUCTUATION: Percent = Percent::from_percent(10);
+const ALICE: AccountId32 = AccountId32::new([1u8; 32]);
+const BOB: AccountId32 = AccountId32::new([2u8; 32]);
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+struct GasFeeDetail {
+    module: String,
+    extrinsic: String,
+    gas_fee: f64,
+}
+
+fn get_call_details(call: &crate::Call) -> (DispatchInfo, u32) {
+    let dispatch_info =
+        <<Runtime as frame_system::Config>::Call as GetDispatchInfo>::get_dispatch_info(call);
+    let call_len = call.using_encoded(|e| e.len()) as u32;
+    (dispatch_info, call_len)
+}
+
+#[test]
+fn diff_gas_fees() {
+    const CURRENT_PATH: &str = env!("CARGO_MANIFEST_DIR");
+    let mut latest_version = String::new();
+    for file in std::fs::read_dir(format!("{CURRENT_PATH}/tx-fees-data")).unwrap() {
+        let _version = file
+            .unwrap()
+            .path()
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .split("-tx-fees.csv")
+            .collect::<Vec<&str>>()[0]
+            .to_string();
+        let _version = version_compare::Version::from(&_version).unwrap();
```

Comment:
this unwrap is failing

Reply:
I just committed a fix, should be fixed.